test/parallel/test-fs-cp.mjs

```diff
@@ -336,6 +336,24 @@ if (!isWindows) {
   }, { code: 'ERR_INVALID_RETURN_VALUE' });
 }
 
+// It should not throw exception if child folder
+// does not pass filter function
+{
+  // mimic there's a file in dest with the same name as the child folder in src
+  // expect: this shouldn't throw error since filtered out by filter function
+  const src = nextdir();
+  mkdirSync(join(src, 'foo'), mustNotMutateObjectDeep({ recursive: true }));
+  
+  const dest = nextdir();
+  mkdirSync(dest, mustNotMutateObjectDeep({ recursive: true }));
+  writeFileSync(join(dest, 'foo'), 'foo-content', mustNotMutateObjectDeep({ mode: 0o444 }));
+  
+  cpSync(src, dest, {
+    filter: path => !path.includes('foo'),
+    recursive: true,
+  });
+}
```

Comment:
Before fixing, when I run the test it throws the following error because `checkPathsSync(srcItem, destItem, opts)` gets called even though the `srcItem` shouldn't be copied:
```
  ---
    SystemError [ERR_FS_CP_DIR_TO_NON_DIR]: Cannot overwrite directory with non-directory: cp returned EISDIR (cannot overwrite directory /Users/xxx/dev/node/test/.tmp.0/copy_28/foo with non-directory /Users/xxx/dev/node/test/.tmp.0/copy_29/foo) /Users/xxx/dev/node/test/.tmp.0/copy_29/foo
        at new SystemError (node:internal/errors:244:5)
        at new NodeError (node:internal/errors:355:7)
        at checkPathsSync (node:internal/fs/cp/cp-sync:77:13)
        at copyDir (node:internal/fs/cp/cp-sync:287:28)
        at onDir (node:internal/fs/cp/cp-sync:268:10)
        at getStats (node:internal/fs/cp/cp-sync:171:12)
        at handleFilterAndCopy (node:internal/fs/cp/cp-sync:158:10)
        at cpSyncFn (node:internal/fs/cp/cp-sync:60:10)
        at cpSync (node:fs:2899:3)
        at file:///Users/xxx/dev/node/test/parallel/test-fs-cp.mjs:353:3 {
      code: 'ERR_FS_CP_DIR_TO_NON_DIR',
      info: {
        message: 'cannot overwrite directory /Users/xxx/dev/node/test/.tmp.0/copy_28/foo with non-directory /Users/xxx/dev/node/test/.tmp.0/copy_29/foo',
        path: '/Users/xxx/dev/node/test/.tmp.0/copy_29/foo',
        syscall: 'cp',
        errno: 21,
        code: 'EISDIR'
      },
      errno: [Getter/Setter],
      syscall: [Getter/Setter],
      path: [Getter/Setter]
    }
```

The reason is 
```
const { destStat } = checkPathsSync(srcItem, destItem, opts);
// check if the path pass the filter function
// 
```